src/nf/nf_conv1d_layer_submodule.f90

```diff
@@ -0,0 +1,183 @@
+    allocate(self % output(self % filters, self % width))
+    self % output = 0
+
+    ! Kernel of shape: filters x channels x kernel_size
+    allocate(self % kernel(self % filters, self % channels, self % kernel_size))
+    call random_normal(self % kernel)
+    self % kernel = self % kernel / self % kernel_size
+
+    allocate(self % biases(self % filters))
+    self % biases = 0
+
+    allocate(self % z, mold=self % output)
+    self % z = 0
+
+    allocate(self % gradient(input_shape(1), input_shape(2)))
+    self % gradient = 0
+
+    allocate(self % dw, mold=self % kernel)
+    self % dw = 0
+
+    allocate(self % db, mold=self % biases)
+    self % db = 0
+
+  end subroutine init
+
+  pure module subroutine forward(self, input)
+    implicit none
+    class(conv1d_layer), intent(in out) :: self
+    real, intent(in) :: input(:,:)
+    integer :: input_channels, input_width
+    integer :: j, n
+    integer :: iws, iwe
+
+    input_channels = size(input, dim=1)
+    input_width    = size(input, dim=2)
+
+    ! Loop over output positions.
+    do j = 1, self % width
+      ! Compute the input window corresponding to output index j.
+      ! In forward: center index = j + half_window, so window = indices j to j+kernel_size-1.
+      iws = j
+      iwe = j + self % kernel_size - 1
+
+      ! For each filter, compute the convolution (inner product over channels and kernel width).
+      do concurrent (n = 1:self % filters)
+        self % z(n, j) = sum(self % kernel(n, :, :) * input(:, iws:iwe))
+      end do
+
+      ! Add the bias for each filter.
+      self % z(:, j) = self % z(:, j) + self % biases
+    end do
+
+    ! Apply the activation function.
+    self % output = self % activation % eval(self % z)
+  end subroutine forward
+
+  pure module subroutine backward(self, input, gradient)
+    implicit none
+    class(conv1d_layer), intent(in out) :: self
+    ! 'input' has shape: (channels, input_width)
+    ! 'gradient' (dL/dy) has shape: (filters, output_width)
+    real, intent(in) :: input(:,:)
+    real, intent(in) :: gradient(:,:)
+
+    integer :: input_channels, input_width, output_width
+    integer :: j, n, k
+    integer :: iws, iwe
+
+    ! Local arrays to accumulate gradients.
+    real :: gdz(self % filters, self % width)  ! local gradient (dL/dz)
+    real :: db_local(self % filters)
+    real :: dw_local(self % filters, self % channels, self % kernel_size)
+
+    ! Determine dimensions.
+    input_channels = size(input, dim=1)
+    input_width    = size(input, dim=2)
+    output_width   = self % width    ! Note: output_width = input_width - kernel_size + 1
+
+    !--- Compute the local gradient gdz = (dL/dy) * sigma'(z) for each output.
+    do j = 1, output_width
+       gdz(:, j) = gradient(:, j) * self % activation % eval_prime(self % z(:, j))
+    end do
+
+    !--- Compute bias gradients: db(n) = sum_j gdz(n, j)
+    do n = 1, self % filters
+       db_local(n) = sum(gdz(n, :), dim=1)
+    end do
```

Comment:
The loop is not needed here IMO
```suggestion
       db_local(:) = sum(gdz(:, :), dim=2)
```
It should be checked if it should be `dim=1` or `dim=2`